staking/programs/staking/src/lib.rs

```diff
@@ -706,4 +706,21 @@ pub mod staking {
             Some(ctx.accounts.config.agreement_hash);
         Ok(())
     }
+
+    /** Recovers a user's `stake account` ownership by transferring ownership
+     * from a token account to the `owner` of that token account.
+     *
+     * This functionality addresses the scenario where a user mistakenly
+     * created a stake account using their token account address as the owner.
+     */
+    pub fn recover_account(ctx: Context<RecoverAccount>) -> Result<()> {
+        let new_owner = ctx.accounts.payer_token_account.owner;
```

Comment:
Can you add a check like the one in line 627 to assert this account has no staked tokens?

Reply:
Transferring accounts with staked tokens might lead to double voting

Reply:
Done